framework/midi_old/event.cpp

```diff
@@ -104,32 +104,6 @@ Event::Event(int t)
     _tuning          = 0.0;
 }
 
-Event::Event(const Event& e)
-    : PlayEvent(e)
-{
-    _type       = e._type;
-    _ontime     = e._ontime;
-    _noquantOntime   = e._noquantOntime;
-    _noquantDuration = e._noquantDuration;
-    _channel    = e._channel;
-    _a          = e._a;
-    _b          = e._b;
-    _duration   = e._duration;
-    _tpc        = e._tpc;
-    _voice      = e._voice;
-    _notes      = e._notes;
-    if (e._edata) {
-        _edata = new unsigned char[e._len + 1];      // don’t forget trailing zero
-        memcpy(_edata, e._edata, e._len + 1);
```

Comment:
If there is a pointer, then the default copy constructor cannot be used (it will just copy the address of the pointer). It's not for nothing that new memory is allocated here and the value is copied.

Example:
```
struct Object
{
    int val = 0;
    int* ptr = nullptr;

    Object()
    {
        ptr = new int[4];
    }

    Object(const Object& o) = default;
};

int main(int argc, char* argv[])
{
    Object o1;
    o1.val = 1;
    o1.ptr[0] = 1;

    Object o2(o1);

    o2.val = 2;
    o2.ptr[0] = 2;

    qDebug() << "o1 val: " << o1.val << ", ptr[0]: " << o1.ptr[0];
    qDebug() << "o2 val: " << o2.val << ", ptr[0]: " << o2.ptr[0];
    qDebug() << "o1 ptr address: " << o1.ptr << ", o2 ptr address: " << o2.ptr;
}

Output:
o1 val:  1 , ptr[0]:  2 <- wrong
o2 val:  2 , ptr[0]:  2
o1 ptr address:  0x55db06c59c90 , o2 ptr address:  0x55db06c59c90
```

Reply:
I realised this pretty quickly and spent about 4 hours trying various syntaxes to try and get GCC to not throw the warning and I swear it's got to do with qt's template thing or something

I can't seem to make sense of why it's throwing a warning like this >:(

```
In file included from /home/aaron/Qt/5.15.2/gcc_64/include/QtCore/qobject.h:49,
                 from /home/aaron/Qt/5.15.2/gcc_64/include/QtCore/qabstractanimation.h:43,
                 from /home/aaron/Qt/5.15.2/gcc_64/include/QtCore/QtCore:6,
                 from /home/aaron/Qt/5.15.2/gcc_64/include/QtGui/QtGuiDepends:3,
                 from /home/aaron/Qt/5.15.2/gcc_64/include/QtGui/QtGui:3,
                 from all.h:62:
/home/aaron/Qt/5.15.2/gcc_64/include/QtCore/qlist.h: In instantiation of ‘void QList<T>::node_copy(QList<T>::Node*, QList<T>::Node*, QList<T>::Node*) [with T = Ms::Event]’:
/home/aaron/Qt/5.15.2/gcc_64/include/QtCore/qlist.h:860:13:   required from ‘QList<T>::QList(const QList<T>&) [with T = Ms::Event]’
/home/aaron/programming/musescore-dev/MuseScore/framework/midi_old/event.h:348:7:   required from here
/home/aaron/Qt/5.15.2/gcc_64/include/QtCore/qlist.h:492:30: warning: implicitly-declared ‘Ms::Event::Event(const Ms::Event&)’ is deprecated [-Wdeprecated-copy]
  492 |                 current->v = new T(*reinterpret_cast<T*>(src->v));
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/aaron/programming/musescore-dev/MuseScore/framework/midi_old/midifile.h:17,
                 from /home/aaron/programming/musescore-dev/MuseScore/mu4/importexport/internal/bb/bb.h:16,
                 from /home/aaron/programming/musescore-dev/MuseScore/mu4/importexport/internal/bb/bb.cpp:14:
/home/aaron/programming/musescore-dev/MuseScore/framework/midi_old/event.h:309:12: note: because ‘Ms::Event’ has user-provided ‘Ms::Event& Ms::Event::operator=(const Ms::Event&)’
  309 |     Event& operator=(const Event& e);
      |            ^~~~~~~~
```

Reply:
Perhaps this is a bug in gcc 9
https://stackoverflow.com/questions/51863588/warning-definition-of-implicit-copy-constructor-is-deprecated/58441689#58441689